pkg/config/process.go

```diff
@@ -70,3 +94,48 @@ func setupProcesses(config Config) {
 	)
 	procBindEnvAndSetDefault(config, "process_config.process_discovery.interval", 4*time.Hour)
 }
+
+var displayProcConfigEnabledWarningOnce sync.Once
+
+// displayProcConfigEnabledDeprecationWarning displays a deprecation warning for process_config.enabled only once.
+// For testing purposes, it returns a bool describing if the message was displayed or not
+func displayProcConfigEnabledDeprecationWarning() (displayed bool) {
+	displayProcConfigEnabledWarningOnce.Do(func() {
+		log.Debug("process_config.enabled is deprecated, use process_config.container_collection.enabled" +
+			" and process_config.process_collection.enabled instead")
+		displayed = true
+	})
+	return
+}
+
+// GetContainerCollectionEnabled retrieves the value of process_config.container_collection.enabled.
+// If process_config.enabled is set, we display a deprecation warning and use that value instead.
+func GetContainerCollectionEnabled(config Config) bool {
```

Comment:
Both `GetContainerCollectionEnabled` and `GetProcessCollectionEnabled` seems to do the same thing, could we create a helper function for both ?

Reply:
Hi, sorry for misunderstanding, but I think these checks are slightly different in the way that they handle reading the legacy `process_config.enabled`. Do you mean that we should return a struct like:

```go
struct {
  containerCollectionEnabled, processAgentEnabled bool
}
```

I think this may reduce the need for us to have two functions, but I don't have a strong opinion either way.

Reply:
I was working on this last night and I came to a solution that combines helpers to a single transform function using `AddOverrideFunc`. Please let me know what you think!